Tools/paletteable sprite editor/js/lib/marvinj-0.9.js

```diff
@@ -0,0 +1,2665 @@
+	this.onload = callback;
+	this.image = new Image();
+	var ref = this;
+	this.image.onload = function(){ref.callbackImageLoaded(ref)};
+	this.image.crossOrigin="anonymous";
+	this.image.src = url;
+};
+
+// WARN: the callback "this" object is the reference to js Image object. 
+MarvinImage.prototype.callbackImageLoaded = function(marvinImage){
+	marvinImage.width = marvinImage.image.width;
+	marvinImage.height = marvinImage.image.height;
+	marvinImage.canvas = document.createElement('canvas');
+	marvinImage.canvas.width = marvinImage.image.width;
+	marvinImage.canvas.height = marvinImage.image.height;
+	
+	
+	marvinImage.ctx = marvinImage.canvas.getContext("2d");
+	marvinImage.ctx.drawImage(marvinImage.image, 0, 0);
+	
+	this.imageData = marvinImage.ctx.getImageData(0, 0, marvinImage.getWidth(), marvinImage.getHeight());
+	
+	if(marvinImage.onload!=null){
+		marvinImage.onload();
+	}
+};
+
+MarvinImage.prototype.clone = function(){
+	var image = new MarvinImage(this.getWidth(), this.getHeight(), this.colorModel);
+	MarvinImage.copyColorArray(this, image);
+	return image;
+};
+
+MarvinImage.prototype.update = function(color){
+	this.canvas.getContext("2d").putImageData(this.imageData, 0,0);
+};
+
+MarvinImage.prototype.clear = function(color){
+	for(var y=0; y<this.getHeight(); y++){
+		for(var x=0; x<this.getWidth(); x++){
+			this.setIntColor(x,y,color);
+		}
+	}
+};
+
+MarvinImage.prototype.getColorModel = function(){
+	return this.colorModel;
+};
+
+MarvinImage.prototype.getAlphaComponent = function(x,y){
+	var start = ((y*this.getWidth())+x)*4; 
+	return this.imageData.data[start+3];
+};
+
+
+MarvinImage.prototype.setAlphaComponent = function(x,y, alpha){
+	var start = ((y*this.getWidth())+x)*4; 
+	this.imageData.data[start+3] = alpha;
+};
+
+MarvinImage.prototype.getIntComponent0 = function(x,y){
+	var start = ((y*this.getWidth())+x)*4; 
+	return this.imageData.data[start];
+};
+
+MarvinImage.prototype.getIntComponent1 = function(x,y){
+	var start = ((y*this.getWidth())+x)*4; 
+	return this.imageData.data[start+1];
+};
+
+MarvinImage.prototype.getIntComponent2 = function(x,y){
+	var start = ((y*this.getWidth())+x)*4; 
+	return this.imageData.data[start+2];
+};
+
+MarvinImage.prototype.setIntColor = function(x,y, a1, a2, a3, a4){
+	if(a2 == null){
+		this.setIntColor1(x,y,a1);
+	} else if(a3 == null && a4 == null){ 
+		this.setIntColor2(x,y,a1,a2);
+	}
+	else if(a4 == null){
+		this.setIntColor3(x,y,a1,a2,a3);
+	}
+	else{
+		this.setIntColor4(x,y,a1,a2,a3,a4);
+	}
+};
+
+MarvinImage.prototype.getIntColor = function(x,y){
+	var start = ((y*this.getWidth())+x)*4;
+	
+	return 	0x100000000 + 
+			(this.imageData.data[start+3] << 24) + 
+			(this.imageData.data[start] << 16) +
+			(this.imageData.data[start+1] << 8) +
+			(this.imageData.data[start+2]);
+};
+
+MarvinImage.prototype.setIntColor1 = function(x,y, color){
+	var a = (color & 0xFF000000) >>> 24;
+	var r = (color & 0x00FF0000) >> 16;
```

Comment:
Codacy found an issue: [The numeric literal '0x00FF0000' will have at different value at runtime.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6395467)